src/directives/formly-field.js

```diff
@@ -87,9 +87,9 @@ function formlyField($http, $q, $compile, $templateCache, $interpolate, formlyCo
       if (!key || !model) {
         return
       }
+
       // If we are working with a number then $parse wont work, default back to the old way for now
-      if (angular.isNumber(key)) {
-        // TODO: Fix this so we can get several levels instead of just one with properties that are numeric
+      if (angular.isNumber(key) || (/\d/.test(key) && key.indexOf('[') === -1)) {
```

Comment:
I agree with pulling the statement out.

In terms of the ^d issue basically if the field has a number in it and it’s not something like this foo[0] then it has to use model[key].

I suppose I could do a check for the first item being numeric but it seemed like overkill to me.

From:  "Kent C. Dodds" <notifications@github.com>
Reply-To:  formly-js/angular-formly <reply@reply.github.com>
Date:  Thursday, December 3, 2015 at 12:37 PM
To:  formly-js/angular-formly <angular-formly@noreply.github.com>
Cc:  Matt Kruczek <kruz79@gmail.com>
Subject:  Re: [angular-formly] fix(formly-field): issue number 566 - Breaking change in 7.3.2 (#569)


In src/directives/formly-field.js:
>        // If we are working with a number then $parse wont work, default back to the old way for now
> -      if (angular.isNumber(key)) {
> -        // TODO: Fix this so we can get several levels instead of just one with properties that are numeric
> +      if (angular.isNumber(key) || (/\d/.test(key) && key.indexOf('[') === -1)) {
Do you suppose that /\d/ could actually be /^\d/? Isn't the number problem only present if the first character of the key is a number? I feel like if we implement this as it is, we'd have a bug where a key like this: w390ekd032s.waky would result in the model getting the property set incorrectly. (same for the test below).

Actually, also, it'd probably be nice to pull this if statement out into a function that both this line and the one below can use to avoid bugs in the future (changing one but not the other)

—
Reply to this email directly or view it on GitHub.